hudi-aws/src/main/java/org/apache/hudi/aws/sync/AWSGlueCatalogSyncClient.java

```diff
@@ -578,6 +578,14 @@ public Map<String, String> getMetastoreSchema(String tableName) {
     }
   }
 
+  @Override
+  public List<FieldSchema> getMetastoreFieldSchemas(String tableName) {
+    Map<String, String> schema = getMetastoreSchema(tableName);
+    return schema.entrySet().stream()
+          .map(f -> new FieldSchema(f.getKey(), f.getValue()))
+          .collect(Collectors.toList());
+  }
```

Comment:
nice catch, is it possible we write a test for it.

Reply:
it needs an integration tests with a glue endpoint. It might be possible with moto. However it's a large task. I can work on it on an other PR. That would make the aws module more stable i hope